x/dispensation/handler_test.go

```diff
@@ -74,6 +75,196 @@ func TestNewHandler_CreateDistribution_MultipleTypes(t *testing.T) {
 	require.NotNil(t, res)
 }
 
+func TestNewHandler_CreateDistribution_PayRewardsInAnyToken_HappyCase(t *testing.T) {
+	app, ctx := test.CreateTestApp(false)
+	keeper := app.DispensationKeeper
+	handler := dispensation.NewHandler(keeper)
+	recipients := 3
+	outputList := test.CreatOutputList(recipients, "10")
+	runner := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address())
+	distributor := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address())
+	totalCoins, err := dispensationUtils.TotalOutput(outputList)
+	assert.NoError(t, err)
+	totalCoins = totalCoins.Add(totalCoins...)
+	err = sifapp.AddCoinsToAccount(types.ModuleName, app.BankKeeper, ctx, distributor, totalCoins)
+	assert.NoError(t, err)
+	// Airdrop distribution type
+	msgAirdrop := types.NewMsgCreateDistribution(distributor, types.DistributionType_DISTRIBUTION_TYPE_AIRDROP, outputList, runner.String())
+	res, err := handler(ctx, &msgAirdrop)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	res, err = handler(ctx, &msgAirdrop)
+	require.Error(t, err)
+	require.Nil(t, res)
+	msgAirdropOutput := msgAirdrop.Output
+	assert.Equal(t, recipients, len(msgAirdropOutput))
+	for i := 0; i < len(msgAirdropOutput); i++ {
+		//(testing) So users should get random catk or ceth coins here.
+		assert.True(t, msgAirdropOutput[i].Coins.AmountOf("catk").Equal(sdk.NewInt(10)) ||
+			msgAirdropOutput[i].Coins.AmountOf("ceth").Equal(sdk.NewInt(10)))
+	}
+	msgLM := types.NewMsgCreateDistribution(distributor, types.DistributionType_DISTRIBUTION_TYPE_LIQUIDITY_MINING, outputList, runner.String())
+	res, err = handler(ctx, &msgLM)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	distributionName := fmt.Sprintf("%d_%s", ctx.BlockHeight(), msgAirdrop.Distributor)
+	msgRun := types.NewMsgRunDistribution(runner.String(), distributionName, types.DistributionType_DISTRIBUTION_TYPE_AIRDROP)
+	res, err = handler(ctx, &msgRun)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	records := keeper.GetRecordsForNameAndStatus(ctx, distributionName, types.DistributionStatus_DISTRIBUTION_STATUS_COMPLETED)
+	assert.Len(t, records.DistributionRecords, 3)
+	records = keeper.GetRecordsForNameAndStatus(ctx, distributionName, types.DistributionStatus_DISTRIBUTION_STATUS_PENDING)
+	assert.Len(t, records.DistributionRecords, (recipients*2)-3)
+
```

Comment:
Can the additional coin balance be asserted here on the receiving addresses?

Reply:
sure.